src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelEventSource.cs

```diff
@@ -79,37 +141,235 @@ public void ConnectionRejected(string connectionId)
         }
 
         [NonEvent]
-        public void RequestStart(HttpProtocol httpProtocol)
+        public void ConnectionQueued(BaseConnectionContext connection)
+        {
+            Interlocked.Increment(ref _connectionQueueLength);
+            if (IsEnabled())
+            {
+                ConnectionQueued(
+                    connection.ConnectionId,
+                    connection.LocalEndPoint?.ToString(),
+                    connection.RemoteEndPoint?.ToString());
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        [Event(6, Level = EventLevel.Informational)]
+        private void ConnectionQueued(string connectionId,
```

Comment:
Should we change this Event Name? As you mentioned offline suffixing with Start/Stop gives semantic meaning to Events?

Reply:
ConnectionQueuedStart?

Reply:
While "Dequeued" is a little more explicit that "Stop", hopefully it will be obvious enough that it's the time in the queue that's being measured, not just the short amount of time it takes to enqueue the connection/request.

Reply:
Will do.